src/ussd_pmtct_seed.js

```diff
@@ -0,0 +1,583 @@
+
+        self.add("state_birth_month", function(name) {
+            return new ChoiceState(name, {
+                question: $("In which month were you born?"),
+                choices: utils.make_month_choices(
+                    $, utils.get_january(self.im.config.testing_today), 12, 1, "MM", "MMM"),
+                next: function(choice) {
+                    self.im.user.set_answer("dob_month", choice.value);
+                    return "state_birth_day";
+                }
+            });
+        });
+
+        self.add("state_birth_day", function(name) {
+            return new FreeText(name, {
+                question: $("Please enter the date of the month you were born (For example 21)"),
+                check: function(content) {
+                    if (utils.is_valid_date(self.im.user.answers.dob_year + '-' +
+                        self.im.user.answers.dob_month + '-' + content, "YYYY-MM-DD")) {
+                        return null;  // vumi expects null or undefined if check passes
+                    } else {
+                        return $("Invalid date. Please enter the date of the month you were born (For example 21)");
+                    }
+                },
+                next: function(content) {
+                    self.im.user.set_answer("dob_day", content);
+                    self.im.user.set_answer("mom_dob",utils.get_entered_birth_date(
+                        self.im.user.answers.dob_year, self.im.user.answers.dob_month, content));
+
+                    return "state_hiv_messages";
+                }
+            });
+        });
+
+        self.add("state_hiv_messages", function(name) {
+            return new ChoiceState(name, {
+                question: $(
+                    "Would you like to receive messages about keeping your child HIV-negative? " +
+                    "The messages will contain words like HIV, medicine & ARVs"),
+                choices: [
+                    new Choice("yes", $("Yes")),
+                    new Choice("no", $("No"))
+                ],
+                next: function(choice) {
+                    if (choice.value === "yes") {
+                        self.im.user.set_answer("hiv_opt_in", "true");
+                        return "state_register_pmtct";
+                    } else {
+                        self.im.user.set_answer("hiv_opt_in", "false");
+                        return "state_end_hiv_messages_declined";
+                    }
+                }
+            });
+        });
+
+        self.add("state_register_pmtct", function(name) {
+            var identity_info = self.im.user.answers.identity;
+            identity_info.details.consent = self.im.user.answers.consent;
+            identity_info.details.mom_dob = self.im.user.answers.mom_dob;
+            identity_info.details.pmtct = {};
+            identity_info.details.pmtct.lang_code = self.im.user.lang || "eng_ZA";
+            identity_info.details.source = "pmtct";
```

Comment:
This should now only be set if this app has created the identity, which it never should since they should have a momconnect subscription to arrive at this point.
We really should move this source setting to the `get_or_create_contact` util, I think.